firedrake/adjoint/blocks.py

```diff
@@ -0,0 +1,108 @@
+from dolfin_adjoint_common.compat import compat
+from dolfin_adjoint_common import blocks
+from pyadjoint.block import Block
+
+
+class Backend:
+    @property
+    def backend(self):
+        import firedrake
+        return firedrake
+
+    @property
+    def compat(self):
+        import firedrake
+        return compat(firedrake)
```

Comment:
fixed

Reply:
It should be the same problem than for the Mesh._coordinates_function comment since the `utils.cached_property` decorator is called on another decorator, shouldn't it?